code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -13,9 +13,9 @@
 /obj/item/weapon/gun/projectile/detective/special_check(var/mob/living/carbon/human/M) //to see if the gun fires 357 rounds safely. A non-modified revolver randomly blows up
 	if(getAmmo()) //this is a good check, I like this check
 		var/obj/item/ammo_casing/AC = loaded[1]
-		if(caliber[POINT38] == 0) //if it's been modified, this is true
+		if(caliber[POINT38] == 1) //unmodified revolver always fires
 			return 1
-		if(istype(AC, /obj/item/ammo_casing/a357) && !perfect && prob(70 - (getAmmo() * 10)))	//minimum probability of 10, maximum of 60
+		if(istype(AC, /obj/item/ammo_casing/a357) &&  (caliber[POINT357] == 1) && !perfect && prob(70 - (getAmmo() * 10)))	//minimum probability of 10, maximum of 60
```

Comment:
I'd rewrite this proc so that it goes through the easiest returns first, since this is combat-live and should process ASAP. Example

```
/obj/item/weapon/gun/projectile/detective/special_check(var/mob/living/carbon/human/M)
if(perfect) //perfectly modified gun always fires correctly
    return 1
var/dakka = getAmmo() //run the semi expensive proc once
if(!dakka) //out of dakka
    return 0
var/obj/item/ammo_casing/AC = loaded[1]
if(!istype(AC, /obj/item/ammo_casing/a357)) //only 357 explodes
    return 1
if(prob(70 - (dakka * 10))) /10% explode with 6 bullets, 60% with 1 (i don't like this balance but that's what it was supposed to be despite never working)
    boom code
else
    return 1
```

And tbh, 20 burn damage and a silent qdel sucks. There's no light explosion, nor any boom noise. special_check triggers before the gun firing sound. Perhaps add a boom noise here, or even a light explosion?